onnxruntime/core/providers/cuda/tensor/split.cc

```diff
@@ -76,6 +76,31 @@ Status SplitKernel::ComputeInternal(OpKernelContext* ctx) const {
   auto input_dims = input_shape.GetDims();
   auto output_dimensions{input_shape.AsShapeVector()};
 
+  if (split_sizes.size() == 3 && (block_size_inside_axis_dim == 1)) {
+    // we use block_size_inside_axis_dim == 1 to check if we are splitting on inner most axis.
```

Comment:
This comment might not be right. block_size_inside_axis_dim == 1 does not mean it is last dimension. For example, input shape is [2, 1, 1] and we split in the first dim.